OpenSim/Common/DelimFileAdapter.h

```diff
@@ -47,11 +93,20 @@ class OSIMCOMMON_API DelimFileAdapter : public FileAdapter {
     DelimFileAdapter(const std::string& delimitersRead,
                      const std::string& delimterWrite);
 
+    /** Create the adapter by setting the delimiters.                         */
+    DelimFileAdapter(const std::string& delimitersRead,
+                     const std::string& delimterWrite,
+                     const std::string& compDelimRead,
+                     const std::string& compDelimWrite);
+
     DelimFileAdapter* clone() const override;
 
     /** Key used for table associative array returned/accepted by write/read. */
     static const std::string _table;
 
+    /** Name of the data type T (template parameter).                         */
+    static const std::string dataTypeName();
+
```

Comment:
Did you consider a generic implementation, like:
```cpp
return SimTK::NiceTypeName<T>::namestr(); 
```
instead of having to specialize below?